### ISSUE TITLE
forbidden to pass list of classes to QueryBuilder

### ISSUE THREAD

**Issue by ltalirz** [aiidateam/aiida_core#2418]
On `provenance_redesign`, I cannot pass a list of classes to the QueryBuilder:

```
In [1]: qb=QueryBuilder()

In [2]: qb.append([Node,Node])
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-e54667174ba5> in <module>()
----> 1 qb.append([Node,Node])

/Users/leopold/Personal/Postdoc-MARVEL/repos/aiida/aiida_rmq/aiida/orm/querybuilder.py in append(self, cls, type, tag, filters, project, subclassing, edge_tag, edge_filters, edge_project, outerjoin, **kwargs)
    568
    569             else:
--> 570                 self._cls_to_tag_map[cls] = tag
    571                 l_class_added_to_map = True
    572

TypeError: unhashable type: 'list'
```

Using a tuple instead seems to work fine
```
In [1]: qb=QueryBuilder()

In [2]: qb.append((Node,Node))
Out[2]: <aiida.orm.querybuilder.QueryBuilder at 0x112105350>
```

Should lists be allowed?
If yes, this should be fixed and tested.
If not, we should change the documentation of `QueryBuilder.append`

**Comment by ltalirz** [comment]
@sphuber agrees that it should be possible to pass a list

**Comment by sphuber** [comment]
Fixed in PR #2421
